### DISCUSSION TITLE
Site.js ❤️ Owncast: one-command / 30 second production installs

### DISCUSSION THREAD

**Discussion by aral** [owncast/owncast#935] [General Owncast discussion]
![Screenshot from 2021-04-14 18-29-21@2x](https://user-images.githubusercontent.com/41798/114754156-28a86080-9d50-11eb-9317-667606414577.png)

So yesterday, I enabled proxy servers in production on [Site.js](https://sitejs.org) so it could be used for Owncast deployments.

Today, I’ve simplified to process of installing and securely running Owncast on a plain vanilla production server to two commands:

1. Install Site.js:
    ```shell
    wget -qO- https://sitejs.org/install | bash
    ```
2. Install and run Owncast:
    ```shell
    site enable --owncast
    ```
3. There is no three!

This is how I just installed Owncast in under 30 seconds for us at Small Technology Foundation at https://owncast.small-tech.org.

## What this does:

  - Install Owncast if it isn’t already installed (the Owncast installer, in turn, will install ffmpeg if it isn’t already installed).
  - Set up Owncast as a systemd service.
  - Set up Site.js as a systemd service.
  - Start serving Owncast at http://localhost:8080 and the chat at ws://localhost:8080
  - Start Site.js as a TLS proxy at https://your.hostname to serve your Owncast instance over HTTPS and WSS.

As usual, your Let’s Encrypt certificates will be automatically provisioned when you first hit your Owncast instance and renewed automatically for you from there on in.

Note that currently, while Site.js will get automatic updates, Owncast will not. However, newer versions of Site.js will always install the latest release version of Owncast. So, to update Owncast, simply disable and re-enable your server using the command above.

## 💕️

Wanted you folks to be the first to know.

[As I mentioned earlier](https://github.com/owncast/owncast/discussions/925), Owncast is a [Small Web](https://small-tech.org/research-and-development) (single-tenant) app and we should embrace what that means in terms of simplicity and the doors that opens. Making installation not 30 minutes but 30 seconds is an orders of magnitude improvement. It means that one day having your own Owncast server could be as easy as signing up for your own Twitch account.

And that’s when we win :)

Lots of love to you and can’t wait to talk about this and more when Gabe’s our guest on next month’s [Small is Beautiful](https://small-tech.org/small-is-beautiful). We also have one tomorrow where we’ll be looking into this week’s National Rail accessibility fail and how it ties into tracking scripts, etc. (and I’ll of course be mentioning this too).

Take care + can’t wait to see where this goes!

**Comment by gabek** [comment]
This is so cool, I'm blown away!  This will be a fantastic option for people don't already have a SSL proxy setup.  I can't wait to play with this tonight and update our documentation to add a dedicated page for installing and running via this method.  People will love it!

**Reply by aral** [reply]
Yay! Please let me know if you have any issues :)

**Reply by obaviet** [reply]
how use domain like you 
![Screenshot 2023-09-14 151001](https://github.com/owncast/owncast/assets/4661721/164375fb-4d3e-4c76-9c46-1f6fea92d213)

**Comment by aral** [comment]
Quick update: I just released Site.js 16.4.1 which should now actually make the single-command installation work on any server without the need to install any other dependencies (previously the installation would fail if curl and unzip were not present on the system).

(I ported the installation script itself to Node.js instead of using the regular bash script.)

**Comment by obaviet** [comment]
my error:
ite enable --owncast
site: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by site)
site: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by site)
site: /lib64/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by site)
[root@instance-106 ~]#

**Comment by gabek** [comment]
@obaviet What version of Owncast are you trying to run?

**Reply by obaviet** [reply]
I simply executed the above command with centos7.  It seems then that it works with ubuntu. However, it doesn't have a reverse proxy, so I haven't been able to run it with the domain yet.
Once installed with the site.js does it have a reverse proxy available to run with the domain?

**Reply by obaviet** [reply]
![Screenshot 2023-09-14 151001](https://github.com/owncast/owncast/assets/4661721/633090e2-46f8-4bdb-95b3-ba2fd64b7b56)

**Reply by aral** [reply]
Hi @obaviet, if you set your hostname (e.g., via `hostnamectl`) to match your domain name, Site.js will automatically pick that up and serve your site over TLS on port 443.

**Reply by obaviet** [reply]
@aral It works great, however Owncast is an old version, how to install to the latest version. I disabled and re-enabled it but nothing changed, it's still version 0.012

**Reply by aral** [reply]
@obaviet Happy to hear you got it working. Apologies about the version; I have to manually update it and my focus is on [Kitten](https://codeberg.org/kitten/app) and [Domain](https://codeberg.org/domain/app) at the moment.

That said, I’ll try my best to make some time this weekend to get round to it. Updating the version is easy; I just need to test that it doesn’t break anything.

**Reply by obaviet** [reply]
I think the current version is also very good, you probably don't need to upgrade it. The new version has a screen when offline that is worse than the current version. For me personally I'm probably more concerned with performance since I'm just pushing owncast onto my other website.
